pkg/go/graph/weighted_graph.go

```diff
@@ -68,6 +81,54 @@ func (wg *WeightedAuthorizationModelGraph) AddEdge(fromID, toID string, edgeType
 	wg.edges[fromID] = append(wg.edges[fromID], edge)
 }
 
+func (wg *WeightedAuthorizationModelGraph) UpsertEdge(fromNode, toNode *WeightedAuthorizationModelNode, edgeType EdgeType, tuplesetRelation string, condition string) error {
+	if fromNode == nil || toNode == nil {
+		return fmt.Errorf("%w: Model cannot be parsed", ErrInvalidModel)
+	}
+
+	if condition == "" {
+		condition = NoCond
+	}
+
+	edges := wg.edges[fromNode.uniqueLabel]
+	for _, edge := range edges {
+		if edge.to.uniqueLabel == toNode.uniqueLabel {
+			if edge.edgeType == edgeType && edge.tuplesetRelation == tuplesetRelation {
+				for _, cond := range edge.conditions {
+					if cond == condition {
+						return nil
+					}
+				}
+				edge.conditions = append(edge.conditions, condition)
+				return nil
+			}
+		}
```

Comment:
cognitive complexity: can reduce one level of nesting
```suggestion
		if edge.to.uniqueLabel == toNode.uniqueLabel && edge.edgeType == edgeType && edge.tuplesetRelation == tuplesetRelation {
			for _, cond := range edge.conditions {
				if cond == condition {
					return nil
				}
			}
			edge.conditions = append(edge.conditions, condition)
			return nil
		}
```